src/averageAges.js

```diff
@@ -55,7 +65,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => (
+    onlyWithSon
+      ? people.find(mother => mother.name === child.mother
+        && child.sex === 'm')
+
+      : people.find(mother => mother.name === child.mother)
```

Comment:
Let's move the `people.find(mother => mother.name === child.mother)` expression to the variable